main.py

```diff
@@ -49,17 +55,28 @@ def validate_id(cls, values):
 products = []
 
 
-# эдитпоин для добавления нового продукта
-@app.post('/products')
-async def add_product(product: Product):
-    try:
-        products.append(product.dict())
-        return {'status': 'success', 'data': product.dict()}, 201
-    except ValidationError as e:
-        return {'status': 'error', 'message': e.errors()}, 400
-
-
-# эдитпоинт для получения списка всех продуктов
 @app.get('/products/all')
 async def get_all_products():
     return {'status': 'success', 'data': products}, 200
+
+
+# Define the endpoint to add a new Product to MongoDB
+@app.post("/products/add")
+async def add_product_to_db(product: ProductCreate):
```

Comment:
You don't have anything asynchronous in your code. You standard synchronous function definition